### ISSUE TITLE
[Browser] JSDom is unable to pull external dependencies when run inside a browser

### ISSUE THREAD

**Issue by czardoz** [tmpvar/jsdom#1285]
I followed the method used [here](https://github.com/tmpvar/jsdom/issues/1018#issuecomment-73269131) to create a browserified jsdom script.

It seems that jsdom is not able to load the ``request`` library properly. This is the error I see in the Chrome browser console:
![screenshot from 2015-11-07 18 32 26](https://cloud.githubusercontent.com/assets/3726484/11015267/f8cfe0c6-857d-11e5-8793-e81f6568b5ac.png)

This the HTML page I loaded:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title></title>
    <script src='jsdom_browserify.js'></script>
    <script>
        window.onload = function () {
            var jsdom = require('jsdom_capsule');
            var virtualConsole = jsdom.createVirtualConsole().sendTo(console);
            jsdom.env({
                html: '<!DOCTYPE html><html><head></head><body></body></html>',
                scripts: [
                    'https://cdnjs.cloudflare.com/ajax/libs/jquery/1.11.2/jquery.min.js',
                    'https://cdnjs.cloudflare.com/ajax/libs/lodash.js/3.6.0/lodash.min.js'
                ],
                done: function (err, window) {
                    console.log('I\'m done');
                },
                features: {
                    FetchExternalResources: ['script', 'link'],
                    ProcessExternalResources: ['script']
                },
                virtualConsole: virtualConsole
            });
        }
    </script>
</head>
<body>

</body>
</html>
```
I guess I could solve the issue by using a custom ``resourceLoader``, but I wonder why this is happening in the first place. The ["browser" directive in the ``package.json``](https://github.com/tmpvar/jsdom/blob/master/package.json#L55-L59) should ideally take care of this, because `request` is being replaced by `browser-request`.

**Comment by Joris-van-der-Wel** [comment]
Actually, the browser directive is working fine. `browser-request` is related to the issue.

The `request` library provides a [`jar()`](https://www.npmjs.com/package/request#request-jar) method which creates a new cookie jar (which jsdom then hackishly wraps so that it can let you specify a custom cookie jar `config.cookieJar`).

However, `browser-request` does not support cookies ([which is not possible in a browser anyway](https://fetch.spec.whatwg.org/#forbidden-header-name)). So jsdom should test if the `jar()` method is present

I'll see if I can whip up a patch

**Comment by czardoz** [comment]
Ah, thanks for the fix!

**Comment by domenic** [comment]
Fixed by #1286; we'll do a new release including that change sometime in the next day or two. Thanks for finding!
